pkg/extract/oracle/oracle.go

```diff
@@ -41,24 +72,457 @@ func Extract(args string, opts ...Option) error {
+			if !ok {
+				return errors.Errorf("no stateref: %s", test.State.StateRef)
+			}
+
+			switch obj.Name {
+			case "oraclelinux-release":
+				switch {
+				case state.Version != nil:
+					for i := range pkgs {
+						pkgs[i].major = strings.TrimPrefix(state.Version.Text, "^")
+					}
+				case state.Arch != nil:
+					for i := range pkgs {
+						pkgs[i].arch = state.Arch.Text
+					}
+				default:
+					return errors.Errorf("invalid combination, obj %+v and state %+v", obj, state)
+				}
+			default:
+				switch {
+				case state.Evr != nil:
+					if state.Evr.Operation != "less than" {
+						return errors.Errorf("unexpected evr operation: %s", state.Evr.Operation)
+					}
+					for i := range pkgs {
+						pkgs[i].name = obj.Name
+						pkgs[i].fixedVersion = state.Evr.Text
+					}
+				case state.SignatureKeyid != nil:
+				case state.Release != nil:
+					// <rpminfo_state xmlns="http://oval.mitre.org/XMLSchema/oval-definitions-5#linux" id="oval:com.oracle.elsa:ste:20163515002" version="501">
+					//   <release operation="pattern match">ksplice</release>
+					// </rpminfo_state>
+					// ksplice information *seems* included in version information, e.g. 2:2.17-106.0.1.ksplice1.el7_2.4
+				default:
+					return errors.Errorf("invalid combination, obj %+v and state %+v", obj, state)
+				}
+			}
+		} else {
+			test, ok := tos.textfileTests[c.TestRef]
+			if !ok {
+				return errors.Errorf("no testref: %s", c.TestRef)
+			}
+			obj, ok := tos.textfileObjs[test.Object.ObjectRef]
+			if !ok {
+				return errors.Errorf("no objectref: %s", test.Object.ObjectRef)
+			}
+			state, ok := tos.textfileStates[test.State.StateRef]
+			if !ok {
+				return errors.Errorf("no stateref: %s", test.State.StateRef)
+			}
+			if !strings.HasPrefix(obj.Filepath.Text, "/etc/dnf/modules.d/") {
+				continue
+			}
+
+			// <ind-def:pattern operation="pattern match">\[container\-tools\][\w\W]*</ind-def:pattern>
+			remaining, found := strings.CutPrefix(obj.Pattern.Text, `\[`)
+			if !found {
+				return errors.Errorf(`unexpected module pattern at prefix. expected: \[, actual: %s`, obj.Pattern.Text)
+			}
+			remaining, found = strings.CutSuffix(remaining, `\][\w\W]*`)
+			if !found {
+				return errors.Errorf(`unexpected module pattern at suffix. expected: \][\w\W]*, actual: %s`, remaining)
+			}
+			module := strings.ReplaceAll(remaining, `\`, "")
+
+			// <ind-def:text operation="pattern match">\nstream\s*=\s*ol8\b[\w\W]*\nstate\s*=\s*(enabled|1|true)|\nstate\s*=\s*(enabled|1|true)[\w\W]*\nstream\s*=\s*ol8\b</ind-def:text>
+			remaining, found = strings.CutPrefix(state.Text.Text, `\nstream\s*=\s*`)
+			if !found {
+				return errors.Errorf(`unexpected stream pattern at prefix. expected: \nstream\s*=\s*, actual: %s`, state.Text.Text)
+			}
+			remaining, _, found = strings.Cut(remaining, `\b[\w\W]`)
+			if !found {
+				return errors.Errorf(`unexpected stream pattern at suffix. expected: \nstream\s*=\s*, actual: %s`, remaining)
+			}
```

Comment:
You're assuming that you always start with `\nstream\s*=\s*`, but why not consider the possibility that it could be in the reverse order, like this: `\nstate\s*=\s*(enabled|1|true)[\w\W]*\nstream\s*=\s*ol8\b|\nstream\s*=\s*ol8\b[\w\W]*\nstate\s*=\s*(enabled|1|true)`

Reply:
```diff
:100644 100644 67a3a5a6 00000000 M	pkg/extract/oracle/oracle.go

diff --git a/pkg/extract/oracle/oracle.go b/pkg/extract/oracle/oracle.go
index 67a3a5a6..c48ce68a 100644
--- a/pkg/extract/oracle/oracle.go
+++ b/pkg/extract/oracle/oracle.go
@@ -383,18 +383,26 @@ func evalCriterions(pkgs []ovalPackage, tos tos, criterions []oracle.Criterion)
 			module := strings.ReplaceAll(remaining, `\`, "")
 
 			// <ind-def:text operation="pattern match">\nstream\s*=\s*ol8\b[\w\W]*\nstate\s*=\s*(enabled|1|true)|\nstate\s*=\s*(enabled|1|true)[\w\W]*\nstream\s*=\s*ol8\b</ind-def:text>
-			remaining, found = strings.CutPrefix(state.Text.Text, `\nstream\s*=\s*`)
-			if !found {
-				return errors.Errorf(`unexpected stream pattern at prefix. expected: \nstream\s*=\s*, actual: %s`, state.Text.Text)
+			var ss []string
+			for _, s := range strings.Split(state.Text.Text, `\n`) {
+				if s == "" {
+					continue
+				}
+
+				lhs, rhs, ok := strings.Cut(s, `\s*=\s*`)
+				if !ok {
+					return errors.Errorf("unexpected pattern. expected: %s, actual: %s", `<entry>\s*=\s*<value>`, s)
+				}
+				if lhs == "stream" {
+					ss = append(ss, strings.ReplaceAll(strings.TrimSuffix(strings.TrimSuffix(strings.TrimSuffix(rhs, `[\w\W]*`), `\b`), "|"), `\`, ""))
+				}
 			}
-			remaining, _, found = strings.Cut(remaining, `\b[\w\W]`)
-			if !found {
-				return errors.Errorf(`unexpected stream pattern at suffix. expected: \nstream\s*=\s*, actual: %s`, remaining)
+			ss = util.Unique(ss)
+			if len(ss) != 1 {
+				return errors.Errorf("stream cannot be determined to a single value. text: %s", state.Text.Text)
 			}
-			// There may be "." in stream and should be unescaped
-			stream := strings.ReplaceAll(remaining, `\`, "")
 			for i := range pkgs {
-				pkgs[i].modularityLabel = fmt.Sprintf("%s:%s", module, stream)
+				pkgs[i].modularityLabel = fmt.Sprintf("%s:%s", module, ss[0])
 			}
 		}
 	}

```

Reply:
Adopted! 88c33ac